ocaml/xapi/xapi_session.ml

```diff
@@ -791,25 +791,38 @@ module Caching = struct
      disabled. This function exists to delay the construction of the
      cache, as Xapi_globs configuration is not guaranteed to have been
      populated before the top-level code of this module is executed. *)
-  let get_or_init_cache () =
-    if not !Xapi_globs.external_authentication_cache_enabled then
+  let get_or_init_cache ~__context =
+    (* read db here, Helpers.get_pool *)
+    let pool = Helpers.get_pool ~__context in
+    let cache_enabled =
+      Db.Pool.get_ext_auth_cache_enabled ~__context ~self:pool
+    in
+    if not cache_enabled then
       None
     else
-      let capacity = !Xapi_globs.external_authentication_cache_size in
+      let capacity =
```

Comment:
Could this be moved into the case in which the cache doesn't exist to save a DB access in most cases?